Os/Linux/SystemResources.cpp

```diff
@@ -82,31 +80,17 @@ namespace Os {
     }
 
     SystemResources::SystemResourcesStatus SystemResources::getMemUtil(MemUtil &memory_util) {
-        FILE *fp = nullptr;
-        NATIVE_INT_TYPE total = 0;
-        NATIVE_INT_TYPE free = 0;
-        // Fallbacks
-        memory_util.total = 1;
-        memory_util.used = 1;
-
-        fp = fopen("/proc/meminfo", "r");
-        if (fp == nullptr) {
-            return SYSTEM_RESOURCES_ERROR;
-        }
-        // No string concerns as strings discarded
-        if (fscanf(fp, "%*s %d %*s", &total) != 1 ||  /* 1st line is MemTotal */
-            fscanf(fp, "%*s %d", &free) != 1) {   /* 2nd line is MemFree */
-            fclose(fp);
-            return SYSTEM_RESOURCES_ERROR;
-        }
-        fclose(fp);
 
-        // Check results
-        if (total < 0 or free < 0 or total < free) {
+        struct sysinfo memory_info;
+        sysinfo(&memory_info);
+        
+        if (memory_info.totalram < memory_info.freeram) {
```

Comment:
@LeStarch Do you think it is still necessary to do this check? I kept it by mimicry with the old code, but in my opinion it's not useful anymore because theoretically it's the `sys_info()` function that takes care of it.

Reply:
I cannot see in the man page where it does this.  Thus I'd keep it because it prevents the subtraction later from going off the rails.

Since we could assume that this is true, we could use an FW_ASSERT, but I think it is ok how it is as there is an establish error-return pattern.